### PR TITLE
SiteSelector: use preferences data component to trigger re-renders.

### PR DESCRIPTION
**Pull request by mtias**
Makes the display of valid data more robust when localStorage gets cleared.

### CODE REVIEW

File: client/lib/sites-list/list.js [hunk 1]
```diff
@@ -38,9 +38,6 @@ function SitesList() {
 	this.selected = null;
 	this.lastSelected = null;
 	this.propagateChange = this.propagateChange.bind( this );
-
-	PreferencesActions.fetch();
-	this.recentlySelected = PreferencesStore.get( 'recentSites' ) || [];
```
Comment (rralian) [inline client/lib/sites-list/list.js]: I'm getting a number of errors that I think have to do with the removal of these lines when SitesList is constructed. Try going to the stats page, then clearing localStorage, and then reloading.

**Comment by rralian** [comment]
I think we need more refactoring before we can remove the preferences fetching from SitesList construction. I see the following errors when loading the stats page with empty localStorage:
stats page

and this too

other error
line error

File: client/lib/sites-list/list.js (hunk 1 above)
Comment (mtias) [inline client/lib/sites-list/list.js]: > I see the following errors when loading the stats page with empty localStorage

@rralian those are present in master and not related to this, it's just how the stats controller is validating the data.

Fixed the other issue during navigation, thanks.

Comment (rralian) [inline client/lib/sites-list/list.js]: Yep, I noticed that and edited my comment. :smile:

Comment (mtias) [inline client/lib/sites-list/list.js]: I did patch the `length` check. :)

**Comment by rralian** [comment]
Code looks good and works fine, though circleci is choking on `make lint` both here and in master.

**Comment by mtias** [comment]
Yeah.

**Pull request merged**